src/nvim/testdir/test_messages.vim

```diff
@@ -65,6 +65,35 @@ func Test_message_completion()
   call assert_equal('"message clear', @:)
 endfunc
 
+func Test_echomsg()
+  call assert_equal("\nhello", execute(':echomsg "hello"'))
+  call assert_equal("\n", execute(':echomsg ""'))
+  call assert_equal("\n12345", execute(':echomsg 12345'))
+  call assert_equal("\n[]", execute(':echomsg []'))
+  call assert_equal("\n[1, 2, 3]", execute(':echomsg [1, 2, 3]'))
+  call assert_equal("\n{}", execute(':echomsg {}'))
+  call assert_equal("\n{'a': 1, 'b': 2}", execute(':echomsg {"a": 1, "b": 2}'))
+  if has('float')
+    call assert_equal("\n1.23", execute(':echomsg 1.23'))
+  endif
+  call assert_match("function('<lambda>\\d*')", execute(':echomsg {-> 1234}'))
+endfunc
+
+func Test_echoerr()
+  throw 'skipped: Nvim does not support test_ignore_error()'
+  call test_ignore_error('IgNoRe')
```

Comment:
what's the point of this? Could maybe use `pcall_err` in a Lua test.

Reply:
`test_ignore_error` was added to suppress errors for a GUI test in patch 8.0.0392 but it is an N/A patch (b659841570d0f1bf51e45ba47ebabc80c84e80ef). It suppress errors so that Vim can execute code that will not be executed in a try/catch/finally. It's like `test_override` so I didn't port the patch.

Reply:
Yes, I don't think we should port it. But maybe it is worth writing a Lua test?